src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -63,6 +63,148 @@ def _durable_async_tool_executor(self, execute_tool_fn):
+            budget = max(0, int(getattr(config, "prefetch_token_budget", 512)))
+            if not limit or not budget:
+                return ""
+            scope = self._memory_prefetch_scope()
+            if hasattr(memory, "search_long_term"):
+                results = memory.search_long_term(query, limit=limit, **scope)
+            elif hasattr(memory, "search"):
+                results = memory.search(query, limit=limit, **scope)
+            else:
+                return ""
+            return self._format_memory_prefetch(results, limit, budget)
+        except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)